File: tickezy_backend/src/models/event.js
const { DataTypes } = require('sequelize');
const sequelize = require('../../db');

const Event = sequelize.define('Event', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  title: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  description: DataTypes.TEXT,
  location: DataTypes.STRING,
  eventDate: DataTypes.DATE,
  price: DataTypes.DOUBLE,
  totalTickets: DataTypes.INTEGER,
  ticketsSold: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
  },
  imageURL: DataTypes.STRING,
  isPublished: {
    type: DataTypes.BOOLEAN,
    defaultValue: false,
  },
  category: {
    type: DataTypes.ENUM('CONCERT', 'SPORTS', 'CONFERENCE', 'THEATER', 'OTHER'),
    defaultValue: 'OTHER',
  },
  status: {
    type: DataTypes.ENUM('UPCOMING', 'ONGOING', 'COMPLETED', 'CANCELLED'),
    defaultValue: 'UPCOMING',
  },
}, {
  timestamps: true,
});

module.exports = Event;

File: TICKEZY/TICKEZY/Models/Event.swift
//
//  Event.swift
//  TICKEZY
//
//  Created by M.A on 10/14/25.
//

import Foundation

enum EventCategory: String, Codable, CaseIterable {
    case CONCERT, SPORTS, CONFERENCE, THEATER, OTHER
}

enum EventStatus: String, Codable, CaseIterable {
    case UPCOMING, ONGOING, COMPLETED, CANCELLED
}


struct Event: Codable, Identifiable {
    let id: String
    let title: String
    let description: String
    let location: String
    let eventDate: Date
    let price: Double
    let totalTickets: Int
    let ticketsSold: Int
    let imageURL: String?       
    let createdBy: String?
    let createdAt: Date?
    let updatedAt: Date?
    let isPublished: Bool
    let category: EventCategory
    let status: EventStatus

    var availableTickets: Int {
        max(0, totalTickets - ticketsSold)
    }
    
    // Computed property to generate full URL
    var fullImageURL: URL? {
        guard let imageURL else { return nil }
        if imageURL.hasPrefix("http") {
            return URL(string: imageURL)
        } else {
            // Replace localhost with your server URL if needed
            return URL(string: "http://localhost:3000\(imageURL)")
        }
    }
}
